modules/@apostrophecms/piece-type/index.js

```diff
@@ -182,7 +182,7 @@ module.exports = {
     },
     async getOne(req, _id) {
       self.publicApiCheck(req);
-      _id = self.apos.i18n.inferIdLocaleAndMode(req, _id);
+      _id = self.apos.i18n.inferIdLocaleAndMode(req, _id, self.name);
```

Comment:
Wouldn't it be pretty easy to get the localization option off the piece type to pass in as a boolean here instead?